### ISSUE TITLE
Make spell-check.R put words in alphabetical order

### ISSUE THREAD

**Issue by cansavvy** [jhudsl/OTTR_Template#233]
**Is your feature request related to a problem? Please describe.**
I know I could add two lines and have spell-check.R do this automatically. It would save us that much more time.

**Comment by cansavvy** [comment]
Also while addressing this, I should make it so the dictionary.txt in _Leanpub doesn't get overwritten by the one in _Bookdown. Simple way is to just change its location, more complicated way is have the merge into one dictionary.

**Comment by kweav** [comment]
Is this still an open issue?

**Comment by cansavvy** [comment]
Yes! I added a step to order the dictionary but it doesn't commit the dictionary back to the branch.

**Comment by kweav** [comment]
Is [this the file that would need to be edited](https://github.com/jhudsl/ottr-reports/blob/main/.github/workflows/main.yml) to add an action to add the dictionary back to the branch? Would keeping the _Bookdown and _Leanpub dictionaries separate be another step within OTTR Template?

**Comment by cansavvy** [comment]
Yes. I think best approach would be adding it to the action.yml of the ottr_reports repo. I have to think about this a bit but feel free to post a plan you're thinking in the mean time.

**Comment by cansavvy** [comment]
Or maybe main.yml is the place? I'll post later after I get a chance to think about it

**Comment by kweav** [comment]
The file where I saw files being committed to a branch was the [`report-maker.yml`](https://github.com/jhudsl/ottr-reports/blob/main/.github/workflows/report-maker.yml), but that seems to be a general action for committing any check output files for whichever check is being run. `action.yml` seemed to be doing the general driving of the running the tests, and the tests themselves seemed to be defined in `main.yml`.  I thought adding a step/action after the [Find outputs](https://github.com/jhudsl/ottr-reports/blob/1ba856c23402cd8fbe8ca548f6c140bda11cddd6/.github/workflows/main.yml#L33) action seemed like where it might fit.
Thank you for thinking about it -- I'll definitely need your guidance on this!

**Comment by cansavvy** [comment]
@kweav I think the strategy might be to have a separate job for this in report-maker.yml

Some clarity for me and you:
- `main.sh` does the crux of the script calling for the error checks and is called by report-maker.yml
- `report-maker.yml` is what is actually called in other repo and takes care of all a lot of the surrounding jobs affiliated with the main error checks. 
- `main.yml` is just for running tests here and won't affect other repos. 

The committing of the dictionary.txt seems like a separate job that is not core to the error checks them so *I think* report-maker.yml is the right place. 

So if we have a new job here https://github.com/jhudsl/ottr-reports/blob/1ba856c23402cd8fbe8ca548f6c140bda11cddd6/.github/workflows/report-maker.yml#L22
That's basically `sort-dictionary` or something like that. It's job would be to retrieve the branch that triggered the pull request that report-maker.yml was called on and commit the dictionary.txt directly to that branch. 

So it would need a 
- Checkout that pull request branch and make sure it has credentials to do so using the `token` argument. 
- Commit the dictionary text file to that branch (not the preview branch like the other jobs). 


The best way to build and test this is to open a pr on another Ottr repo and call the branch you are working from by changing one of the error calls to the branch you are working from in ottr-reports 

For example in a pull_request.yml file in an OTTR report you'd use a `@branchname` instead of the `@main` that's standard so like:

`uses: jhudsl/ottr-reports/.github/workflows/report-maker.yml@branch_name`

When you are ready to test you'd push to that additional branch in that OTTR repo. You wouldn't merge that repo's PR you'd just use it for testing your ottr-reports dev. 

Let me know if this makes sense. It was helpful for me to write some of it down. Thanks for taking in interest in this! Either way would be great to have you work on some of the issues on OTTR reports if you're game!

**Comment by kweav** [comment]
Thanks, @cansavvy! This clarity and strategy is super helpful and I think I can move forward with building/testing and will let you know if I have questions
